Tests/UnitTests/Mocks/MockDeviceCache.swift

```diff
@@ -221,4 +221,18 @@ class MockDeviceCache: DeviceCache {
         invokedClearLatestNetworkAndAdvertisingIdsSentParameters = (appUserID, ())
         invokedClearLatestNetworkAndAdvertisingIdsSentParametersList.append((appUserID, ()))
     }
+
+    var invokedSetLatestNetworkAndAdvertisingIdsSent = false
+    var invokedSetLatestNetworkAndAdvertisingIdsSentCount = 0
+    var invokedSetLatestNetworkAndAdvertisingIdsSentParameters: (appUserID: String?, Void)?
+    var invokedSetLatestNetworkAndAdvertisingIdsSentParametersList = [(appUserID: String?, Void)]()
+
+    override func set(latestNetworkAndAdvertisingIdsSent: [String: String], appUserID: String) {
+        invokedSetLatestNetworkAndAdvertisingIdsSent = true
+        invokedSetLatestNetworkAndAdvertisingIdsSentCount += 1
+        invokedSetLatestNetworkAndAdvertisingIdsSentParameters = (appUserID, ())
+        invokedSetLatestNetworkAndAdvertisingIdsSentParametersList.append((appUserID, ()))
+        super.set(latestNetworkAndAdvertisingIdsSent: latestNetworkAndAdvertisingIdsSent, appUserID: appUserID)
```

Comment:
Does this need to call the real implementation? I don't think it's expected for the mock to update the underlying data.

Reply:
yeah i did this to test that the postIfNeeded would skip if something was already set... suppose i can just use some booleans in MockDeviceCache though